### ISSUE TITLE
Fix weekly email report

### ISSUE THREAD

**Issue by paolodamico** [PostHog/posthog#1937]
Fix release from #1860 (hotfix #1934).

- Cron job to send weekly email report is processed multiple times. Review scheduling process to send to only one worker.
- Implement mechanism to ensure each email is sent only once to each recipient.
- Improve the way the async task is processed by creating a separate task for each organization, instead of processing all organizations in a single task.
